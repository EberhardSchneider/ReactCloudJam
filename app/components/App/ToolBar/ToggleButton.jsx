import React from 'react';
import PropTypes from 'prop-types';



export default class ToggleButton extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      active: false
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    e.preventDefault();
    this.setState((prevState) => {
      return {
        active: !prevState.active
      };
    });
  }


  render() {
    const {
      text
    } = this.props;

    const toggleStyle = this.state.active ? {
      color: '#fff',
      backgroundColor: 'green'
    } : {
      color: '#000',
      backgroundColor: 'white'
    };

    return (
      <button
        className="toggle-button button-primary"
        style={toggleStyle}
        onClick={this.handleClick}>
        {text}
      </button>
    );
  }
}

ToggleButton.propTypes = {
  text: PropTypes.string
};