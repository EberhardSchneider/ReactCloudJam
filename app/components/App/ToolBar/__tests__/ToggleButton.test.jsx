import React from 'react';
import renderer from 'react-test-renderer';

import ToggleButton from '../ToggleButton';

beforeEach(() => {

});

test('button changes style when clicked', () => {
  const component = renderer.create(
    <ToggleButton />
  );
  let tree = component.toJSON();
  const styleBeforeClick = tree.props.style;

  tree.props.onClick();

  tree = component.toJSON();
  const styleAfterClick = tree.props.style;

  expect(styleBeforeClick).not.toEqual(styleAfterClick);
});

test('button should be the same as last time', () => {
  const component = renderer.create(
    <ToggleButton />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});