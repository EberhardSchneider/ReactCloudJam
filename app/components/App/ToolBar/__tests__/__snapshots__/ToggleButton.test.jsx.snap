// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should be the same as last time 1`] = `
<button
  className="toggle-button button-primary"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "color": "#000",
    }
  }
/>
`;
